server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -274,6 +370,48 @@ private Consumer findConsumer(String consumerUuid) {
         return consumer;
     }
 
+    /**
+     * Returns the pool object that is identified by the given id, if it is found in the system.
+     * Otherwise, it throws a NotFoundException.
+     *
+     * @param poolId the id of the pool we are searching for.
+     *
+     * @return the pool that was found in the system based on the given id.
+     *
+     * @throws NotFoundException
+     *  if the pool with the given id was not found in the system.
+     */
+    private Pool findPool(String poolId) {
+        Pool pool = poolManager.find(poolId);
```

Comment:
I'd need to check, but this might fail in when poolId is null or empty. If so we need to wrap this (and other lookups in similar methods) in a null or empty check.